src/backend/utils/adt/arrayfuncs.c

```diff
@@ -945,6 +945,8 @@ CopyArrayEls(ArrayType *array,
 	int			bitmask = 1;
 	int			i;
 
+	Insist(values);
```

Comment:
I think you should instead rewrite the conditions here somehow: https://github.com/arenadata/gpdb/blob/fd5939d4f491238bbd28be88b98949cd0e2d6fe0/src/backend/utils/adt/arrayfuncs.c#L3063-L3073

Reply:
Why? `values` must not be null in `CopyArrayEls`, so I think it's better to add this condition there

Reply:
1) elems = NULL
2) fixedwidthtype = false
3) call CopyArrayEls with NULL as values
may be condition should be
```c
if (elems==NULL || fixedwidthtype)
{
/* do nothing */
}
```
?

Reply:
It seems like here we allocate a new array to fill it with CopyArrayEls, so I think your proposal is correct
https://github.com/arenadata/gpdb/blob/fd5939d4f491238bbd28be88b98949cd0e2d6fe0/src/backend/utils/adt/arrayfuncs.c#L3021-L3042

Reply:
I think it should be like 
```
if (elems==NULL && fixedwidthtype)
{
/* do nothing */
} else if (elems != NULL) {
    CopyArrayEls
```
However, the program shouldn't run any further if elems == NULL and !fixedwidthtype so it may be also `Insist`

Reply:
failed tests: uao_dml/uao_dml_row uao_dml/uao_dml_column

Reply:
I think we should just use Insist in else block

Reply:
> if (elems==NULL && fixedwidthtype)
> {
> /* do nothing */
> } else if (elems != NULL) {
>     CopyArrayEls

this solution is working!

Reply:
But what would happen if elems == NULL and !fixedwidthtype? Doesn't seem like it can be tolerated